utils/s2n_random.c

```diff
@@ -332,8 +341,84 @@ S2N_RESULT s2n_get_private_random_bytes_used(uint64_t *bytes_used)
     return S2N_RESULT_OK;
 }
 
-static int s2n_rand_urandom_impl(void *ptr, uint32_t size)
+S2N_RESULT s2n_rand_get_dev_urandom(struct s2n_rand_device **device)
+{
```

Comment:
I'm torn on whether this is more or less hacky than including the .c file :P

Reply:
I changed this to include the .c file. If it looks more hacky this way let me know and I'll put it back.

Reply:
Unfortunately including the .c file broke the new shared library test from https://github.com/aws/s2n-tls/pull/4370, so I had to revert this. But I tried to make it as least-hacky as possible. Let me know if you see anything I can do to make it better!